src/v/config/configuration.cc

```diff
@@ -120,6 +120,18 @@ configuration::configuration()
       "Interval for which all coprocessor offsets are flushed to disk",
       {.visibility = visibility::tunable},
       300000ms) // five minutes
+  , topic_memory_per_partition(
+      *this,
+      "topic_memory_per_partition",
+      "Required memory per partition when creating topics",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      1_MiB)
+  , topic_fds_per_partition(
+      *this,
+      "topic_fds_per_partition",
+      "Required file handles per partition when creating topics",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      10)
```

Comment:
I see 10 mentioned in the PR description, but why 10?

Reply:
A typical partition uses at least 3 FDs, and will use several (for the vaguest definition of "several") more FDs when doing compaction, or if reads are happening to segments other than the most recent.

10 is a rough order of magnitude.  On a GKE node we get a ulimit of 1M by default, so this pretty permissive in terms of the resulting partition count.  It kicks in as a limiting factor in situations like a naive default desktop linux with nfiles=1024.